src/java/main/org/apache/zookeeper/server/FinalRequestProcessor.java

```diff
@@ -430,6 +432,7 @@ public void processRequest(Request request) {
             // the client and leader disagree on where the client is most
             // recently attached (and therefore invalid SESSION MOVED generated)
             cnxn.sendCloseSession();
+            request.checkLatency();
```

Comment:
Yes. The percentiles and whatsoever should be a matter of another issue. No doubt about this, IMO.

The meat of my comment was in fact if checkLatency isn't in the responsibility of Stats class, even it remaining as an proper method (i.e, no inclusion into updateLatency).